### ISSUE TITLE
Consider adding parameter 'realm' to WWW-Authenticate header to comply with RFC-7617

### ISSUE THREAD

**Issue by andreimilto** [terreng/simple-web-server#153]
[RFC-7617 Section 2](https://datatracker.ietf.org/doc/html/rfc7617#section-2) states that in the Basic authentication scheme the parameter `realm` is required:
```
The Basic authentication scheme utilizes the Authentication Framework as follows.
In challenges:
o  The scheme name is "Basic".
o  The authentication parameter 'realm' is REQUIRED ([RFC7235], Section 2.2).
```

Currently Simple Web Server doesn't use this parameter. If Basic authentication fails, it responds with `401` error and the header:  
`WWW-Authenticate: Basic`

Consider including the parameter `realm` into the `WWW-Authenticate` header to comply with the RFC:  
`WWW-Authenticate: Basic realm="My Realm"`

You may also want to add a corresponding input field in the UI that would allow users to modify the value of this parameter. Users should be able to enter only ASCII characters in that field (see the realm encoding problem - [RFC-7617 Section 3](https://datatracker.ietf.org/doc/html/rfc7617#section-3)).

**Comment by terreng** [comment]
I think we should just set the value to something. Don't think we need an option for it.

**Comment by ethanaobrien** [comment]
>I think we should just set the value to something. Don't think we need an option for it.

What would you like me to set it to

**Comment by terreng** [comment]
@andreimilto MDN says the realm directive is optional. Thoughts? Is there a situation where it's actually needed, or is this solely to comply with the spec?

<img width="802" alt="Screenshot 2023-08-10 at 8 57 07 PM" src="https://github.com/terreng/simple-web-server/assets/11605395/e6dd75f2-e3b3-48a3-8209-002882734574">

**Comment by andreimilto** [comment]
> MDN says the realm directive is optional. Thoughts?

Created an [issue](https://github.com/mdn/content/issues/28531) in the MDN repository and also [asked](https://stackoverflow.com/questions/76848487) Stack Overflow community - hope it makes things clearer.
Personally, I'm more inclined to trust RFC.

> Is there a situation where it's actually needed, or is this solely to comply with the spec?

Haven't encountered a scenario where missing `realm` would break things, so it is primarily to comply with the specs :neckbeard:

**Comment by hamishwillee** [comment]
MDN docs were wrong - realm is required, reflecting the spec - thanks so much for taking the time to post @andreimilto. 

As to whether it is needed, my understanding is that a browser may choose to use this string to decide whether to re-prompt a user for credentials or use stored values. So "probably".

**Comment by terreng** [comment]
Thanks so much, @hamishwillee and @andreimilto!

Seeing as the HTTP Basic Auth option applies globally for all paths, I think it makes sense to just pick a value, like "SimpleWebServer". Advanced configuration could be left up to plugins.

@ethanaobrien Could you make it so?

**Comment by terreng** [comment]
Landed in 1.2.9.
